### PR TITLE
#277 verify amountPlanned

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes: https://github.com/commercetools/commercetools-adyen-integration/issues/277

### CODE REVIEW

File: extension/src/paymentHandler/payment-handler.js [hunk 1]
```diff
@@ -12,7 +12,9 @@ async function handlePayment (paymentObject) {
     // if it's not adyen payment, ignore the payment
```
Comment (ahmetoz) [inline extension/src/paymentHandler/payment-handler.js:12]: not related to this branch, but could you remove lodash dependency from that file if we are not using it.
` error  '_' is assigned a value but never used`

File: extension/src/validator/error-messages.js [hunk 2]
```diff
@@ -15,5 +15,6 @@ module.exports = {
   GET_PAYMENT_METHODS_REQUEST_INVALID_JSON: 'getOriginKeysRequest does not contain valid JSON.',
   MAKE_PAYMENT_REQUEST_INVALID_JSON: 'makePaymentRequest does not contain valid JSON.',
   // eslint-disable-next-line max-len
-  SUBMIT_ADDITIONAL_PAYMENT_DETAILS_REQUEST_INVALID_JSON: 'submitAdditionalPaymentDetailsRequest does not contain valid JSON.'
+  SUBMIT_ADDITIONAL_PAYMENT_DETAILS_REQUEST_INVALID_JSON: 'submitAdditionalPaymentDetailsRequest does not contain valid JSON.',
+  AMOUNT_PLANNED_INVALID: 'amountPlanned field cannot be changed after makePayment request is made to Adyen'
```
Comment (ahmetoz) [inline extension/src/validator/error-messages.js:19]: it's valid but immutable, maybe change the naming in here to 
"AMOUNT_PLANNED_MODIFICATION_IS_NOT_ALLOWED"

Also, the message here is for the user, could we say more precisely: 

`Modifying the "amountPlanned" value is not allowed after the payment amount charged by Adyen.`

File: extension/src/validator/validator-builder.js [hunk 3]
```diff
@@ -20,6 +21,23 @@ function withPayment (paymentObject) {
           = errorMessages.SUBMIT_ADDITIONAL_PAYMENT_DETAILS_REQUEST_INVALID_JSON
       return this
     },
+    validateAmountPlanned () {
```
Comment (ahmetoz) [inline extension/src/validator/validator-builder.js:24]: The function is not tested directly, but indirectly covered in payment-handler tests, why?

File: extension/src/validator/validator-builder.js [hunk 4]
```diff
@@ -20,6 +21,23 @@ function withPayment (paymentObject) {
...
+      const oldMakePaymentRequestObj = paymentObject.interfaceInteractions
```
Comment (ahmetoz) [inline extension/src/validator/validator-builder.js:25]: is the oldest or newest ?

Comment (lojzatran) [reply extension/src/validator/validator-builder.js:25]: It's the newest from the old ones 😄

File: extension/src/validator/validator-builder.js (hunk 3 above)
Comment (lojzatran) [reply extension/src/validator/validator-builder.js:24]: Why not? https://github.com/commercetools/commercetools-adyen-integration/pull/283/commits/b30637e690c0d66bfd4f3ba43523bb090a5fd507

File: extension/src/validator/error-messages.js (hunk 2 above)
Comment (lojzatran) [reply extension/src/validator/error-messages.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/283/commits/fd30c79d4afcf8cbb87739bb7a3d268700fc9e51

I will not change the message to your message as it's probably not true - amountPlanned is not charged in some cases, etc. you need to do the redirect and then verification etc. 
And the real reason for this error is simply after there's a makePayment request, amountPlanned cannot be changed.

File: extension/src/paymentHandler/payment-handler.js (hunk 1 above)
Comment (lojzatran) [reply extension/src/paymentHandler/payment-handler.js:11]: https://github.com/commercetools/commercetools-adyen-integration/pull/283/commits/ddde5580e2e422d302a20bc0195f3a9149b58a1f

**Pull request merged**